src/main/java/com/google/devtools/build/lib/analysis/platform/PlatformUtils.java

```diff
@@ -84,38 +83,44 @@ public static Platform getPlatformProto(
       return null;
     }
 
-    Map<String, String> properties;
+    Map<String, String> properties = new HashMap<>();
     if (!spawn.getCombinedExecProperties().isEmpty()) {
       // Apply default exec properties if the execution platform does not already set
       // exec_properties
       if (spawn.getExecutionPlatform() == null
           || spawn.getExecutionPlatform().execProperties().isEmpty()) {
-        properties = new HashMap<>();
         properties.putAll(defaultExecProperties);
         properties.putAll(spawn.getCombinedExecProperties());
       } else {
         properties = spawn.getCombinedExecProperties();
       }
-    } else if (spawn.getExecutionPlatform() != null
-        && !Strings.isNullOrEmpty(spawn.getExecutionPlatform().remoteExecutionProperties())) {
+    } else if (spawn.getExecutionPlatform() != null) {
       properties = new HashMap<>();
```

Comment:
I don't think you need to re-create the `properties` here.

Reply:
True. This was a code refactor leftover. Removed in c3fa67e090.